### PR TITLE
fix #17190 `nimscript` now accepts arbitrary file extensions

### LINKED ISSUE #17190: Is there a technical reason why nimscript files require the .nims extension?
### Summary

Remove the requirement for nimscript files to have an extension

### Description

Basically every other interpreted language can be use with a #! and no file extension, it would be nice if I want to embed a single nimscript file in a non-nim project and have it appear as any other script.

### Additional Information

After discussing this in the nim discord for a while it appears there is no reason why there HAS to be an extension on nimscript files. This could be wrong though and any additional information on the choice to have it be required is welcome.

### PR DESCRIPTION
**Pull request by ringabout**
fix #17190 `nimscript` now accepts arbitrary file extensions

### CODE REVIEW

File: tests/tools/tnimscriptwithnimext.nim [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+macro foobar(): void =
+  result = newCall(bindSym"echo", newLit("nothing"))
+
+echo "foobar"
+
+let x = 123
+
+foobar()
+
+exec "echo hallo"
```
Comment (ringabout) [inline tests/tools/tnimscriptwithnimext.nim]:
```suggestion
let a = 11
doAssert a == 11
```

File: tests/tools/tnimscriptwithnimext.nim [hunk 2]
```diff
@@ -0,0 +1,22 @@
+discard """
+cmd: "nim e $file"
+output: '''
+foobar
+nothing
+hallo
+'''
```
Comment (ringabout) [inline tests/tools/tnimscriptwithnimext.nim]:
```suggestion
```

File: tests/tools/tnimscriptwithmacro.nims [hunk 3]
```diff
@@ -4,6 +4,7 @@ output: '''
 foobar
 nothing
 hallo
+'''
```
Comment (ringabout) [inline tests/tools/tnimscriptwithmacro.nims]:
```suggestion
```

File: compiler/main.nim [hunk 4]
```diff
@@ -342,8 +342,6 @@ proc mainCommand*(graph: ModuleGraph) =
     if conf.projectIsCmd or conf.projectIsStdin: discard
     elif not fileExists(conf.projectFull):
       rawMessage(conf, errGenerated, "NimScript file does not exist: " & conf.projectFull.string)
-    elif not conf.projectFull.string.endsWith(".nims"):
```
Comment (timotheecour) [inline compiler/main.nim:345]: this doesn't work:
`nim e foo.nimfake` doesn't seem to run nimscript on foo.nimfake.
=> looks like a test is needed; see trunner.nim which is an option when it doesn't fit well with testament spec

also, ideally this would work too: `nim r main.nims`, but can be done later

Comment (ringabout) [reply compiler/main.nim:345]: nim e foo.nimfake

what's the error messages?

Comment (timotheecour) [reply compiler/main.nim:345]: no error message, but nothing runs (eg if it contains `echo "foo"`, it won't print foo)

Comment (ringabout) [reply compiler/main.nim:345]: i see

File: changelog.md [hunk 5]
```diff
@@ -277,6 +277,7 @@
 - Tuple expressions are now parsed consistently as
   `nnkTupleConstr` node. Will affect macros expecting nodes to be of `nnkPar`.
 
+- `nimscript` now accepts arbitrary file extensions.
```
Comment (timotheecour) [inline changelog.md:280]: IMO we should now also deprecate `nim main` ? (without any any command nor `--eval`)
it's not particularly useful when `nim e main` works, and can now cause ambiguities.

should be done in a separate PR though.
